### PR TITLE
cargo-lambda: init at 0.11.2

### PR DESCRIPTION
**Pull request by taylor1791**
###### Description of changes

This adds [cargo-lambda](https://www.cargo-lambda.info/) to nixpkgs. A Cargo subcommand to help you work with AWS Lambda.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable: 
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-lambda/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/cargo-lambda --prefix PATH : ${lib.makeBinPath [ cargo-watch ]}
+  '';
+
+  checkFlags = [
+    "--skip test_download_example"
+    "--skip test_target_dir_from_env"
```
Comment (taylor1791) [inline pkgs/development/tools/rust/cargo-lambda/default.nix:24]: Both of these tests fail. The first accesses the network. Nix's build methodology violates the assumptions of the second.

File: pkgs/development/tools/rust/cargo-lambda/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-IK4HVj8Y8Vz+mza8G9C+m5JRfNT3BWWdlbQQkJPu6RI=";
+  };
+
+  cargoSha256 = "sha256-oSqoSvv8IiChtduQQA31wItHcsnRBAQgOCrQN4sjcx8=";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/cargo-lambda --prefix PATH : ${lib.makeBinPath [ cargo-watch ]}
```
Comment (taylor1791) [inline pkgs/development/tools/rust/cargo-lambda/default.nix:19]: `cargo-watch` is a runtime dependency of  `cargo-lambda`. By ensuring that cargo-watch is available, it will not try and download `cargo-watch` automatically at runtime.

**Comment by taylor1791** [comment]
I am cloning this onto a mac to see if `Security` is the only missing buildInputs.

**Comment by taylor1791** [comment]
I was able to successfully build this on darwin after adding Security.

**Comment by taylor1791** [comment]
Thanks for catching that @calavera!

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -13249,6 +13249,12 @@
     githubId = 102685;
     name = "Thomas Friese";
   };
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:13251]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/development/tools/rust/cargo-lambda/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-IK4HVj8Y8Vz+mza8G9C+m5JRfNT3BWWdlbQQkJPu6RI=";
+  };
+
+  cargoSha256 = "sha256-oSqoSvv8IiChtduQQA31wItHcsnRBAQgOCrQN4sjcx8=";
+
+  buildInputs = []
+    ++ lib.optionals stdenv.isDarwin [ Security ];
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/cargo-lambda/default.nix:17]:
```suggestion
    buildInputs = lib.optionals stdenv.isDarwin [ Security ];
```

File: pkgs/development/tools/rust/cargo-lambda/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/rust/cargo-lambda/default.nix:27]: Please add a short comment to each

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (taylor1791) [reply maintainers/maintainer-list.nix:13307]: I move this into its own commit. As for sorting it alphabetically, I believe it was already sorted correctly. Is it sorted by something other than the attribute name?

File: pkgs/development/tools/rust/cargo-lambda/default.nix (hunk 1 above)
Comment (taylor1791) [reply pkgs/development/tools/rust/cargo-lambda/default.nix:28]: Done.

File: pkgs/development/tools/rust/cargo-lambda/default.nix (hunk 4 above)
Comment (taylor1791) [reply pkgs/development/tools/rust/cargo-lambda/default.nix]: Done.
